Add events ModifyInfoModuleContentEvent, ProvideStaticVariablesToComponentEvent and RenderComponentEvent to event list (#6465)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ApiOverview/Events/Events/Fluid/RenderComponentEvent.rst

..  include:: /Includes.rst.txt
..  index::
    Events; RenderComponentEvent
..  _RenderComponentEvent:

====================
RenderComponentEvent
====================

..  versionadded:: 14.1

The :php-short:`\TYPO3\CMS\Fluid\Event\RenderComponentEvent` can be used to alter or
replace the rendering of `Fluid components <https://docs.typo3.org/permalink/t3coreapi:using-fluid-components>`_.
There are three possible use cases:

> 1.  fully take over the rendering of components by filling the :php:`$renderedContent` with

Reviewer: ```suggestion
1.  Fully take over the rendering of components by filling the :php:`$renderedContent` with
```

it is a sentence, start with a capital letter.

>     :php:`$event->setRenderedContent()`. The first event that does this skips all following

Reviewer: ```suggestion
    :php:`$event->setRenderedContent()`. The first event listener that does this skips all following
```

    event listeners.

> 2.  provide additional arguments (= variables in the component template) or slots to

Reviewer: ```suggestion
2.  Provide additional arguments (= variables in the component template) or slots to
```

    the component with :php:`$event->setArguments()`/:php:`$event->setSlots()`.

> 3.  execute additional code that doesn't influence the component rendering directly, e. g.

Reviewer: ```suggestion
3.  Execute additional code that doesn't influence the component rendering directly, for example
```

We want to avoid those abbreviations.
